packages/connection-storage/src/connection-storage.spec.ts

```diff
@@ -75,6 +87,238 @@ describe('ConnectionStorage', function () {
     }
   });
 
+  describe('migrateToSafeStorage', function () {
+    let sandbox: Sinon.SinonSandbox;
+    beforeEach(function () {
+      sandbox = Sinon.createSandbox();
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    context('does not migrate connections', function () {
+      it('when there are no connections', async function () {
+        await ConnectionStorage.migrateToSafeStorage();
+        const connections = await ConnectionStorage.loadAll();
+        expect(connections).to.deep.equal([]);
+      });
+
+      it('when there are only legacy connections', async function () {
+        await writeFakeConnection(tmpDir, connection1270);
+
+        const encryptSecretsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'encryptSecrets'
+        );
+        const getKeytarCredentialsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'getKeytarCredentials'
+        );
+
+        await ConnectionStorage.migrateToSafeStorage();
+
+        expect(
+          encryptSecretsSpy.called,
+          'it does not try to encrypt'
+        ).to.be.false;
+        expect(
+          getKeytarCredentialsSpy.called,
+          'it does not try to get secrets from keychain'
+        ).to.be.false;
```

Comment:
TIL! Didn't know before that we could add custom message in `expect` for the failure cases. Thanks for this ⭐

Reply:
yeah, makes it easier to understand the expectation when there're many.